pkg/source/grpc/batch.go

```diff
@@ -164,8 +164,11 @@ func (bc *batchChain) run() {
 			return
 		case b := <-bc.batchChan:
 			bs[b.index] = b
-			for _, e := range b.events {
-				bc.productFunc(e)
+			for i := int32(0); i < b.eventIndex; i++ {
+				e, ok := b.events[i]
+				if ok {
+					bc.productFunc(e)
+				}
```

Comment:
This code can be considered for simplification to the following form:

```go
if e, ok := b.events[i]; ok {
	bc.productFunc(e)
}
```

Reply:
Great. Clean code!  Done